edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/ExperimentalFeatures.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Service
+{
+    using Microsoft.Azure.Devices.Edge.Storage;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.Logging;
+
+    public class ExperimentalFeatures
```

Comment:
>ExperimentalFeatures [](start = 17, length = 20)

Should this be called FeatureConfiguration or FeaturesFilter? When they are enabled after a while they are actually not experimental anymore

Reply:
Maybe we can use a common class for edge agent and edge hub, I see some of the features are common.

---
In reply to: [308914144](https://github.com/Azure/iotedge/pull/1464#discussion_r308914144) [](ancestors = 308914144)

Reply:
The reason it was called ExperimentalFeatures_* is to convey the intent that these are experimental features and subject to change in the future. The naming was critical for that. 
I realize that it is going to be a bit tricky to update the flags once the features are not experimental anymore. But I have some thoughts related to what we can do then...

Reply:
As for a common class - the common functionality is by chance, imo, and I don't see there being much overlap between EdgeHub and EdgeAgent going forward.. so slight duplication is probably okay there.